src/ai/susi/server/api/cms/SkillSlideshowService.java

```diff
@@ -0,0 +1,102 @@
+/**
+ *  APIKeysService
+ *  Copyright by Shubham Gupta @fragm3
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ai.susi.server.api.cms;
+
+import ai.susi.DAO;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.json.JsonTray;
+import ai.susi.server.*;
+import ai.susi.server.Authorization;
+import org.json.JSONObject;
+
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This Servlet gives a API Endpoint to add, modify and delete different API
+ * keys used by SUSI. It requires user role to be ADMIN or above ADMIN example:
+ * For adding/uploading new image: Necessary parameters : access_token,
+ * pathname(for navigation on click), image_name, Optional parameter: info
+ * http://localhost:4000/cms/skillSlideshow.json?access_token=go2ijgk5ijkmViAac2bifng3uthdZ&pathname=REDIRECT_URL&info=Metadata&image_name=NAME_OF_IMAGE
+ * For deleting, pathname -> string is necessary and deleteSlide=true
+ * http://localhost:4000/aaa/apiKeys.json?pathname=REDIRECT_URL&deleteSlide=true
+ */
+
+public class SkillSlideshowService extends AbstractAPIHandler implements APIHandler {
+
+    private static final long serialVersionUID = 5000658108778105134L;
+
+    @Override
+    public UserRole getMinimalUserRole() {
+        return UserRole.ADMIN;
+    }
+
+    @Override
+    public JSONObject getDefaultPermissions(UserRole baseUserRole) {
+        return null;
+    }
+
+    @Override
+    public String getAPIPath() {
+        return "/cms/skillSlideshow.json";
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query call, HttpServletResponse response, Authorization authorization,
+            final JsonObjectWithDefault permissions) throws APIException {
+        if (call.get("pathname", null) == null || call.get("image_name", null) == null) {
+            throw new APIException(400, "Bad Request. No enough parameter present");
+        }
+
+        String pathname = call.get("pathname", null);
```

Comment:
Pathname should not be entered by user.

Reply:
Pathname is redirect link, which rename it to redirect_link, to avoid confusion